docs/content/Caching/Caching.md

```diff
@@ -190,6 +124,77 @@ cube(`Orders`, {
 });
 ```
 
+## In-memory Cache
+
+Cube.js caches the results of executed queries using in-memory cache. The cache
+key is a generated SQL statement with any existing query-dependent
+pre-aggregations.
+
+Upon receiving an incoming request, Cube.js first checks the cache using this key.
+If nothing is found in the cache, the query is executed in the database and the result set
+is returned as well as updating the cache.
+
+If an existing value is present in the cache and the `refreshKey` value for
+the query hasn't changed, the cached value will be returned. Otherwise, a
+[query renewal](#in-memory-cache-force-query-renewal) will be performed.
+
+### Refresh Keys
+
+Cube.js takes great care to prevent unnecessary queries from hitting your database. The first stage caching system caches query results in Redis (or in the in-memory store in development), but Cube.js needs a way to know if the data powering that query result has changed. If the underlying data isn't any different, the cached result is valid and can be returned skipping that expensive query, but if there is a difference, the query needs to be re-run and its' result cached.
```

Comment:
```suggestion
Cube.js takes great care to prevent unnecessary queries from hitting your database. The first stage caching system caches query results in Redis (or in the in-memory store in development), but Cube.js needs a way to know if the data powering that query result has changed. If the underlying data isn't any different, the cached result is valid and can be returned skipping an expensive query, but if there is a difference, the query needs to be re-run and its result cached.
```